diff --git a/org/apache/felix/cm/impl/ConfigurationImpl.java b/org/apache/felix/cm/impl/ConfigurationImpl.java
index 9ecde3002fafc..1b74117b1fc4c 100644
--- a/org/apache/felix/cm/impl/ConfigurationImpl.java
+++ b/org/apache/felix/cm/impl/ConfigurationImpl.java
@@ -383,6 +383,8 @@ public class ConfigurationImpl extends ConfigurationBase
             // finally assign the configuration for use
             configure( newProperties );
 
+            getConfigurationManager().removeConfiguration(this);
+
             // if this is a factory configuration, update the factory with
             // do this only after configuring with current properties such
             // that a concurrently registered ManagedServiceFactory service
@@ -633,3 +635,4 @@ public class ConfigurationImpl extends ConfigurationBase
         properties.remove( ConfigurationAdmin.SERVICE_BUNDLELOCATION );
     }
 }
+/* @generated */
\ No newline at end of file
diff --git a/org/apache/felix/cm/impl/ConfigurationManager.java b/org/apache/felix/cm/impl/ConfigurationManager.java
index e9e4620814abc..4d73bcf6a353c 100644
--- a/org/apache/felix/cm/impl/ConfigurationManager.java
+++ b/org/apache/felix/cm/impl/ConfigurationManager.java
@@ -20,6 +20,7 @@ package org.apache.felix.cm.impl;
 
 
 import java.io.IOException;
+import java.lang.reflect.Method;
 import java.security.SecureRandom;
 import java.text.MessageFormat;
 import java.util.ArrayList;
@@ -1247,8 +1248,18 @@ public class ConfigurationManager implements BundleActivator, BundleListener
         Object log = logTracker.getService();
         if ( log != null )
         {
-            ( ( LogService ) log ).log( getServiceReference(), level, message, t );
-            return;
+            Class<?> clazz = log.getClass();
+
+            try {
+                Method method = clazz.getDeclaredMethod(
+                        "log", Integer.class, String.class, Throwable.class);
+
+                method.invoke(getServiceReference(), level, message, t);
+
+                return;
+            }
+            catch (Exception exception) {
+            }
         }
 
         // Otherwise only log if more serious than the configured level
@@ -2058,4 +2069,4 @@ public class ConfigurationManager implements BundleActivator, BundleListener
         }
     }
 }
-
+/* @generated */
